src/app/components/login/login.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import {
+  FormControl,
+  FormGroup,
+  FormGroupDirective,
+  NgForm,
+  Validators,
+} from '@angular/forms';
+import { ErrorStateMatcher } from '@angular/material/core';
+import { AuthService } from 'src/app/services/auth-service.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss'],
+})
+export class LoginComponent {
+  loginForm: any = FormGroup;
+  username = new FormControl('', [
+    Validators.required,
+    Validators.minLength(4),
+  ]);
+
+  password = new FormControl('', [Validators.required]);
+  matcher = new MyErrorStateMatcher();
+
+  constructor(public router: Router, public authService: AuthService) {}
+
+  ngOnInit() {
+    this.authService.userIsLogged = false;
+  }
+
+  loginIn(user: any, pass: any) {
+    user = this.username.value;
+    pass = this.password.value;
+
+    this.authService.login(user, pass);
+  }
+}
+
+export class MyErrorStateMatcher implements ErrorStateMatcher {
+  isErrorState(
+    control: FormControl | null,
+    form: FormGroupDirective | NgForm | null
+  ): boolean {
+    const isSubmitted = form && form.submitted;
+    return !!(
```

Comment:
'!!' are probably not necessary because you are returning the result of the condition. Condition always return boolean